pkg/workflows/wasm/host/module.go

```diff
@@ -110,7 +110,17 @@ type ModuleConfig struct {
 	Determinism *DeterminismConfig
 }
 
-type Module struct {
+type Module interface {
+	Start()
+	Close()
+	IsLegacyDAG() bool
+	// V1/Legacy API - request either the Workflow Spec or Custom-Compute execution
+	Run(ctx context.Context, request *wasmpb.Request) (*wasmpb.Response, error)
+	// V2/"NoDAG" API - request either the list of Trigger Subscriptions or launch workflow execution
+	Execute(ctx context.Context, request *wasmpb.ExecuteRequest) (*wasmpb.ExecutionResult, error)
```

Comment:
@bolekk I would recommend not exposing the protos directly at the interface level -- I started moving away from this with eg. the FetcherFn changes I made the other week.

The reason for this is that the protos are a lot more rigid and unchangeable the moment you start running actual workflows (you don't want to break them); by isolating the protos so that they're an internal detail here you can easily change the interface without introducing a breaking change at the proto level. This is a lot more difficult if you expose the protos directly.

Reply:
This is an interface between Engine and Host, so shouldn't be too hard to change internally IMO, even if it's a breaking change.
Can you elaborate on the recommendation? Do you mean having equivalent Go structs? That feels like unnecessary overhead converting back and forth.

Reply:
If you don't mind, I'd rather keep it for now. We can avoid the extra overhead of converting structs and always migrate the Engine-Host part later when we see it's necessary.